src/main/java/uk/gov/companieshouse/controller/DissolutionController.java

```diff
@@ -83,4 +87,37 @@ public DissolutionGetResponse getDissolutionApplication(
                 .get(companyNumber)
                 .orElseThrow(DissolutionNotFoundException::new);
     }
+
+    @Operation(summary = "Patch Dissolution Application", tags = "Dissolution")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Dissolution Application patched", content = @Content),
+            @ApiResponse(responseCode = "404", description = "Dissolution Application not found"),
+            @ApiResponse(responseCode = "409", description = "Dissolution Request already approved by this director", content = @Content)
+    })
+    @PatchMapping
+    @ResponseStatus(HttpStatus.OK)
+    public DissolutionPatchResponse patchDissolutionApplication(
+            @RequestHeader("ERIC-identity") String userId,
+            @RequestHeader("ERIC-Authorised-User") String authorisedUser,
+            @PathVariable("company-number") final String companyNumber,
+            HttpServletRequest request) {
+
+        if (StringUtils.isBlank(userId) || StringUtils.isBlank(authorisedUser)) {
+            throw new UnauthorisedException();
+        }
+
+        if (!dissolutionService.doesDissolutionRequestExistForCompany(companyNumber)) {
+            throw new DissolutionNotFoundException();
+        }
+
+        final String userEmail = getEmail(authorisedUser);
+
+        if (dissolutionService.hasDirectorAlreadyApproved(companyNumber, userEmail)) {
+            throw new ConflictException();
+        }
```

Comment:
Hmmm now that I think about this, I think this is the wrong way of validating this. The API definition says:
```
The email must match one of the directors who are yet to sign
```

This current check ensures that they haven't already approved the application, but it doesn't ensure that they are one of the defined signatories.

We should only be allowed to provide emails that are yet to sign, something like:
```
if (!dissolutionService.isDirectorPendingApproval(companyNumber, userEmail)) {
  throw new BadRequestException();
}
```